lib/config/config-loader.js

```diff
@@ -166,39 +237,26 @@ async function loadConfigFile(filePath) {
      *
      * When Node.js supports native TypeScript imports, we can remove this check.
      */
-    if (isTS && !isDeno && !isBun) {
 
-        // eslint-disable-next-line no-use-before-define -- `ConfigLoader.loadJiti` can be overwritten for testing
-        const { createJiti } = await ConfigLoader.loadJiti().catch(() => {
-            throw new Error("The 'jiti' library is required for loading TypeScript configuration files. Make sure to install it.");
-        });
+    if (isTS) {
+        if (hasUnstableNativeNodeJsTSConfigFlag) {
+            if (isNativeTypeScriptSupportEnabled()) {
+                return await dynamicImportConfig(filePath, fileURL, mtime);
+            }
 
-        // `createJiti` was added in jiti v2.
-        if (typeof createJiti !== "function") {
-            throw new Error("You are using an outdated version of the 'jiti' library. Please update to the latest version of 'jiti' to ensure compatibility and access to the latest features.");
+            throw new Error(
+                "The unstable_native_nodejs_ts_config flag is enabled, but native TypeScript support is not enabled in the current Node.js process. You need to either enable native TypeScript support by passing --experimental-strip-types or remove the unstable_native_nodejs_ts_config flag."
```

Comment:
I think this error message could be confusing if someone is using a version of Node.js that doesn't support TypeScript natively like Node.js 20.x. In that case it would be better to state in the error message that `unstable_native_nodejs_ts_config` isn't supported in older versions of Node.js.

If `process.features.typescript` is undefined, Node.js has no native TypeScript support.